### PR TITLE
Modal: Close button displays top left if headerText is not provided

### LINKED ISSUE #2598: Modus Modal: Close button displays top left if headerText is not provided
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

The title pretty much sums it up. Since headerText is optional, I wouldn't expect the close 'x' to behave differently, and it is a bit jarring. 

### Reduced test cases

_No response_

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Chrome

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

^0.28.0-react18

### Priority

Low

### What product/project are you using Modus Components for ?

WorksOS / Machine Playback

### What is your team/division name ?

CCFS/Weka

### Are you willing to contribute ?

Maybe

### Are you using Modus Web Components in production ?

_No response_

### PR DESCRIPTION
**Pull request by prashanthr6383**
## Description

References 

Fixes #2598 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by jewel-shajan — COMMENTED** [review]
Testing is finished, and everything looks good. Now moving forward for PR review.

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 1]
```diff
@@ -191,7 +191,7 @@ export class ModusModal {
   renderModalHeader(): JSX.Element[] {
     return (
       <header class={{ scrollable: this.isContentScrollable }}>
-        {this.headerText}
+        <span class="modal-title">{this.headerText}</span>
```
Comment (saravananrmsmy) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:194]: @prashanthr6383 
If our intention is only to keep the close icon at the right can we do it without the introduction of span. Adding `margin-left: auto` for the .header-buttons should work. Can we try it out ?

**Review by saravananrmsmy — CHANGES_REQUESTED** [review]

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx (hunk 1 above)
Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx]: @saravananrmsmy  yes it works great. I have updated

**Review by jewel-shajan — COMMENTED** [review]
Testing has been successfully completed after the changes and the update has been moved to PR review.

**Review by saravananrmsmy — APPROVED** [review]

**Review by coliff — APPROVED** [review]

**Pull request merged**